mkosi/machine.py

```diff
@@ -122,9 +142,46 @@ def run(self, commands: Sequence[str], timeout: int = 900, check: bool = True) -
 
         return process
 
-    def __exit__(self, *args: Any, **kwargs: Any) -> None:
+    def _kill(self, *args: Any, **kwargs: Any) -> None:
         if self._serial:
             self._serial.kill(signal.SIGTERM)
             self.exit_code = self._serial.wait()
             self._serial = None
         self.stack.__exit__(*args, **kwargs)
+
+    def __exit__(self) -> None:
+        self._kill(*exc_info())
+
+    def reboot(self) -> None:
+        self._kill(*exc_info())
+        self._boot()
```

Comment:
I'm aware of [this comment](https://github.com/systemd/mkosi/pull/919#discussion_r815975681). But this looks cleaner since we don't override anything with the `MkosiMachineTest` class. 

Plus the signature for both `exit` and `reboot` looks simpler. 
Unless there's a specific reason to do it that way? Then, please, let me know,

Reply:
Well, this apparently breaks the Machine when used as a context manager.
```
def test_after_shutdown() -> None:
        with Machine(["boot"]) as m:
>           pass
E           TypeError: __exit__() takes 1 positional argument but 4 were given
```

Reply:
This works for all scenarios:
`def __exit__(self, *args: Any, **kwargs: Any) -> None:
        self._kill(*exc_info())

    def reboot(self) -> None:
        self._kill(*exc_info())
        self._boot()`

Reply:
Why not just do:

```
def __exit__(self, *args: Any, **kwargs: Any) -> None:
    self._kill(*args, **kwargs)
```

?

exc_info() should only be used as a last resort since it will only give you the exception that's being processed at the moment. If __exit__ is called after the exception handling block ends, exc_info() will return (None, None, None) which isn't all that useful if the actual exception information is being passed as arguments to `__exit__()`.

See https://docs.python.org/3/library/sys.html#sys.exc_info for more information